api/kessel/relations/v1beta1/check.proto

```diff
@@ -62,4 +71,36 @@ message CheckForUpdateResponse {
 	}
 	Allowed allowed = 1;
 	ConsistencyToken consistency_token = 2;
-}
\ No newline at end of file
+}
+
+message CheckBulkRequestItem {
+	ObjectReference resource = 1 [(buf.validate.field).required = true];
+	string relation = 2 [(buf.validate.field).string.min_len = 1];
+	SubjectReference subject = 3 [(buf.validate.field).required = true];
+}
+
+message CheckBulkResponseItem {
+	enum Allowed {
+		ALLOWED_UNSPECIFIED = 0;
+		ALLOWED_TRUE = 1;
+		ALLOWED_FALSE = 2;
+		// e.g.  ALLOWED_CONDITIONAL = 3;
+	}
+	Allowed allowed = 1;
+}
+
+message CheckBulkResponsePair {
+	CheckBulkRequestItem request = 1;
+	CheckBulkResponseItem item = 2;
+
+}
+
+message CheckBulkRequest {
+	Consistency consistency = 1;
+	repeated CheckBulkRequestItem items = 2 [(buf.validate.field).required = true];
```

Comment:
I'm not sure if  `(buf.validate.field).required = true];` works for a repeated field, might need to use `[(buf.validate.field).repeated.min_items = 1];` like we use for `ImportBulkTuplesRequest`

source: https://buf.build/bufbuild/protovalidate/docs/main:buf.validate#buf.validate.RepeatedRules